src/plugins/no_data_page/public/lib/use_has_api_keys.ts

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import type { HttpSetup } from '@kbn/core-http-browser';
+import { useEffect, useState } from 'react';
+import { NO_DATA_API_PATHS } from '../../common';
+import { HasApiKeysResponse } from '../types';
+
+/**
+ * Creation function for the "useHasApiKeys" React hook.
+ *
+ * Usage:
+ * const MyComponent: React.FC = () => {
+ *   const { hasApiKeys, loading, error } = useHasApiKeys();
+ *
+ *   // Handle error
+ *   throwIfError(error);
+ *
+ *   // Handle loading
+ *
+ *   // Present custom UX for users depending on their access to API Keys
+ * };
+ *
+ * TODO: consider moving this to a package under the Security domain
+ * as it is tied to a Security API endpoint.
+ */
+export const createUseHasApiKeys = ({ http }: { http: HttpSetup }) => {
+  return function useHasApiKeys(): HasApiKeysResponse {
```

Comment:
I'll let you make a call on this but I think this makes more sense as a small service or class instead of a hook. Plain js tends to be more composable and easier to consume by non-React code. Also, hooks tend to introduce extra render cycles.

Reply:
I agree with your suggestion to give this another look and see about a simple, non-React way. Considering this module could move around to the Security domain as need arises, this should be implemented in a flexible way.

>  Also, hooks tend to introduce extra render cycles.

The way I would propose to implement this would be to use an Observable. In that situation, consumer will typically use `useObservable` with this, which is also a hook. I'm not sure if the way I had planned to do this would reduce extra render cycles. Let's see when I have something working.